src/Board.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import './style.css';
+import Squares from './Square';
+// import Snapshot from './Snapshot';
+import {TurnContext, SetTurnContext} from './Game';
+
+function Board() {
+    const [boardState, setBoardState] = React.useState((new Array(9).fill(null)));
+    const [stages, setStages] = React.useState([]);
+    const turn = React.useContext(TurnContext);
+    const setTurn = React.useContext(SetTurnContext);
+    let arr = [...boardState];
+    console.log(arr);
+
+    function declareWinner(boardState) {
+        const conditions = [
+            [0, 1, 2],
+            [3, 4, 5],
+            [6, 7, 8],
+            [0, 3, 6],
+            [1, 4, 7],
+            [2, 5, 8],
+            [0, 4, 8],
+            [2, 4, 6],
+        ];
+
+        for (let i = 0; i < conditions.length; i++) {
+            const [first, second, third] = conditions[i];
+
+            if (boardState[first] && boardState[second] === boardState[first] && boardState[third] === boardState[first]) {
+                return boardState[first];
+            }
+        }
+        return null;
+    }
+
+    const winner = declareWinner(boardState);
+
+    let status;
+    if (winner) {
+        status = `The winner is ${winner}`;
+    }
+
+
+    function clickHandler(num) {
+        if (!winner && !boardState[num]) {
+            const stagesCopy = [...stages];
+            const boardCopy = [...boardState];
+            
+            boardCopy[num] = turn;
+            stagesCopy.push(boardCopy);
+            
+            setBoardState(boardCopy);
+            setStages(stagesCopy);
+
+            setTurn(() => {
+                if (turn === 'X') {
+                    return 'O';
+                }
+                return 'X';
+            });
+        }
+    }
+
+    function moveToStage(step) {
+        const boardInHistory = [...stages];
+        setBoardState(boardInHistory[step]);
+    }
+
+    function goToStart() {
+        setBoardState(new Array(9).fill(null));
+        setStages([]);
+    }
+
+    const differentStages = stages.map((stage) => {
+        const desc = `Go to move #${stage+1}`;
+        return (
+            <li key={stage}>
+                <button onClick={() => moveToStage(stage)}>{desc}</button>
```

Comment:
This seems to be where the bug in the time travel is coming from. What does your `stages` array look like here? Once you confirm that, I bet you will be able to debug this :-)